test/hermes/intl/get-canonical-locales.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// RUN: %hermes -O -target=HBC %s | %FileCheck --match-full-lines %s
+
+print("get canonical locales test");
+// CHECK-LABEL: get canonical locales test
+
+print(Intl.getCanonicalLocales(["EN-us", "Fr"]));
+// CHECK-NEXT: en-US,fr
+
+print(Intl.getCanonicalLocales(["zh-zh", "ZH"]));
+// CHECK-NEXT: zh-ZH,zh
+
+print(Intl.getCanonicalLocales(["aam", "zyb", "art-lojban"]));
+// CHECK-NEXT: aas,za,jbo
+
+print(Intl.getCanonicalLocales([]).length);
+//CHECK-NEXT: 0
+
+try {
+  Intl.getCanonicalLocales([""]);
+  print("Succeeded");
+} catch (e) {
+  print("Caught", e.name, e.message);
+}
+//CHECK-NEXT: Caught{{.*}}
+
+// Invalid unicode
+try {
+  Intl.getCanonicalLocales(["\ud800bc"]);
+  print("Succeeded");
+} catch (e) {
+  print("Caught", e.name, e.message);
+}
+//CHECK-NEXT: Caught{{.*}}
```

Comment:
These should no longer throw an exception right? So the test would fail.

Reply:
Yes, both tests now fail, here is their output.
``` 
>> Intl.getCanonicalLocales(["\ud800bc"]);
[ "", [length]: 1 ]
>> Intl.getCanonicalLocales([""]);
[ "", [length]: 1 ]
```

Reply:
In that case, they should be removed, because they will fail when you run `check-hermes`